Source/Details/ASContentSizeCategory.h

```diff
@@ -0,0 +1,49 @@
+//
+//  ASContentSizeCategory.h
+//  Texture
+//
+//  Copyright (c) 2017-present, Pinterest, Inc.  All rights reserved.
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+
+#import <UIKit/UIKit.h>
+#import <AsyncDisplayKit/ASBaseDefines.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+/**
+ * ASContentSizeCategory is a UIContentSizeCategory that can be used in a struct.
+ */
+typedef NS_ENUM(NSInteger, ASContentSizeCategory) {
```

Comment:
Let's do something clever, and use `__unsafe_unretained UIContentSizeCategory` as a struct member:

```objc
typedef struct {
    // unretained is required because ARC can't manage struct memory
    __unsafe_unretained UIContentSizeCategory category;
    BOOL flag;
} ASTestStruct;
```